p2p/protocol/ping/ping_legacy.go

```diff
@@ -0,0 +1,113 @@
+package ping
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"time"
+
+	u "github.com/ipfs/go-ipfs-util"
+	inet "github.com/libp2p/go-libp2p-net"
+	peer "github.com/libp2p/go-libp2p-peer"
+)
+
+const pingTimeout = 60 * time.Second
+
+// PingHandlerLegacy handles a ping/1.0.0 ping.
+func (ps *PingService) PingHandlerLegacy(s inet.Stream) {
+	buf := make([]byte, PingSize)
+
+	errCh := make(chan error, 1)
+	defer close(errCh)
+	timer := time.NewTimer(pingTimeout)
+	defer timer.Stop()
+
+	go func() {
+		select {
+		case <-timer.C:
+			log.Debug("ping timeout")
+		case err, ok := <-errCh:
+			if ok {
+				log.Debug(err)
+			} else {
+				log.Error("ping loop failed without error")
+			}
+		}
+		s.Reset()
+	}()
+
+	for {
+		_, err := io.ReadFull(s, buf)
+		if err != nil {
+			errCh <- err
+			return
+		}
+
+		_, err = s.Write(buf)
+		if err != nil {
+			errCh <- err
+			return
+		}
+
+		timer.Reset(pingTimeout)
+	}
+}
+
+// PingLegacy pings a peer using ping/1.0.0
+func (ps *PingService) PingLegacy(ctx context.Context, p peer.ID) (<-chan time.Duration, error) {
```

Comment:
Would it make sense to explicitly refer to the version number? If we end up with a v3 at some point, the term "legacy" would become ambiguous. I much rather call the files `ping_v1.go`, `ping_v2.go`, as well as the functions. That also reduces cognitive load on the reader, as it's one less association they need to make (legacy == v1).